plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationContext.hs

```diff
@@ -1,29 +1,119 @@
+-- by omitting the generation of costs concerning the missing builtins.
+clearV3builtinCostModel, clearV2builtinCostModel
+    :: (m ~ BuiltinCostModelBase MCostingFun) => m -> m
+clearV3builtinCostModel r = r
+               { paramBls12_381_G1_add = mempty
+               , paramBls12_381_G1_neg = mempty
+               , paramBls12_381_G1_scalarMul = mempty
+               , paramBls12_381_G1_equal = mempty
+               , paramBls12_381_G1_hashToGroup = mempty
+               , paramBls12_381_G1_compress = mempty
+               , paramBls12_381_G1_uncompress = mempty
+               , paramBls12_381_G2_add = mempty
+               , paramBls12_381_G2_neg = mempty
+               , paramBls12_381_G2_scalarMul = mempty
+               , paramBls12_381_G2_equal = mempty
+               , paramBls12_381_G2_hashToGroup = mempty
+               , paramBls12_381_G2_compress = mempty
+               , paramBls12_381_G2_uncompress = mempty
+               , paramBls12_381_millerLoop = mempty
+               , paramBls12_381_mulMlResult = mempty
+               , paramBls12_381_finalVerify = mempty
+               , paramKeccak_256 = mempty
+               , paramBlake2b_224 = mempty
+               }
+clearV2builtinCostModel r = r
+               { paramSerialiseData = mempty
+               , paramVerifyEcdsaSecp256k1Signature = mempty
+               , paramVerifySchnorrSecp256k1Signature = mempty
+               }
+
+-- Assign to `mempty` those CEK construct that the specific plutus version X introduces, and thus
+-- are missing from version X-1.
+-- Calling this function on a vX_CekMachineCosts will turn it into a vX-1_CekMachineCosts,
+-- by omitting the generation of costs concerning the missing CEK constructs.
+clearV3machineCostModel :: (m ~ CekMachineCostsBase Maybe) => m -> m
+clearV3machineCostModel r = r
+    { cekConstrCost  = mempty
+    , cekCaseCost    = mempty
+    }
 
-Note: It always reflects the latest, in-development Plutus version.
--}
-costModelParamsForTesting :: Plutus.CostModelParams
-costModelParamsForTesting = fromJust Plutus.defaultCostModelParams
+-- Helpers
 
-{- | only for testing purposes: get an evaluation context by re-applying the
-extracted default cost model parameters (resulting in no change to params).
-An alternative implementation would be to pass an empty map of params.
+-- A variant of `extractCostModelParams` that returns the `CostModelParams` not in alphabetical order,
+-- but in the `ParamName` order, i.e. the order expected by the ledger
+extractCostModelParamsLedgerOrder :: (Common.IsParamName p, Ord p, ToJSON machinecosts, ToJSON builtincosts)
+                     => CostModel machinecosts builtincosts
+                     -> Maybe (Map.Map p Integer)
+extractCostModelParamsLedgerOrder cm =
+    let mapInAlphaOrder = extractCostModelParams cm
+        mapInLedgerOrder = Map.mapKeys (fromJust . readParamName) <$> mapInAlphaOrder
```

Comment:
the fromJust seems bad, we're already returning a maybe, can't we fold that in?

Reply:
ok good idea

Reply:
I hope I can find a way to fold this, because `mapKeys` interface seems to be pure